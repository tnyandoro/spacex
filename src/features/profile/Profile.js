import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { loadMissions } from '../missions/missionsReducer';
import { loadRockets } from '../rockets/rocketsReducer';
import './profile.css';

const Profile = () => {
  const dispatch = useDispatch();
  const store = useSelector(({ missionsReducer, rocketsReducer }) => (
    [missionsReducer, rocketsReducer]));
  useEffect(() => {
    if (!store[0].length && !store[1].length) {
      dispatch(loadMissions());
      dispatch(loadRockets());
    }
  }, []);

  const [missions, rockets] = store;
  return (
    <div className="profile-holder">
      <div className="profile-item">
        <h2 className="profile-title">My Mission</h2>
        <div className="ul-profile">
          {missions.map(mission => (
            mission.reserved ? <div key={mission.id} className="li-profile">{mission.name}</div> : null
          ))}
        </div>
      </div>
      <div className="spacer" />
      <div className="profile-item">
        <h2 className="profile-title">My Rockets</h2>
        <div className="ul-profile">
          {rockets.map(rocket => (
            rocket.reserved ? <div key={rocket.id} className="li-profile">{rocket.rocket_name}</div> : null
          ))}
        </div>
      </div>
    </div>
  );
};

export default Profile;
