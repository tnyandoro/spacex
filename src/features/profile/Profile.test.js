import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Profile from './Profile';

describe('Missions Component', () => {
  const initialState = {
    rocketsReducer: [],
    missionsReducer: [],
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);

  it('Profile snapshot test', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
