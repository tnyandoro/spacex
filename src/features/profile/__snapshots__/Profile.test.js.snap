// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Component Profile snapshot test 1`] = `
<div
  className="profile-holder"
>
  <div
    className="profile-item"
  >
    <h2
      className="profile-title"
    >
      My Mission
    </h2>
    <div
      className="ul-profile"
    />
  </div>
  <div
    className="spacer"
  />
  <div
    className="profile-item"
  >
    <h2
      className="profile-title"
    >
      My Rockets
    </h2>
    <div
      className="ul-profile"
    />
  </div>
</div>
`;
